### ISSUE TITLE
SSL Task

### ISSUE THREAD

**Issue by JhuoW** [Xi-yuanWang/GLASS#4]
Hi, 

In the paper, you propose to use node-level, edge-level, and subgraph-level SSL tasks to pre-train the node features. In GNNEmb.py, I can only find the link prediction task. Does it mean edge-level SSL plays a more important role than the other two SSL? 

Besides, in the link prediction, in this [line](https://github.com/Xi-yuanWang/GLASS/blob/6f865380b340f38833ff27fdf03a1257fb6ac26b/GNNEmb.py#L119), you use all positive edges (i.e., edge_index of the original graph) to train the node embeddings, and then use a part of positive+negative edges as a validation set. Thus, the training edge set and validation edge set may be overlapped, right?  

Thank you

**Comment by Xi-yuanWang** [comment]
Hi,

Sorry for my late reply. 

*"Does it mean edge-level SSL plays a more important role than the other two SSL?"*
Yes. We only use edge-level SSL in our experiments.

*"The training edge set and validation edge set may be overlapped, right?"*
Yes. As the task is pretraining rather than link prediction, data leakage prevention is not necessary.

**Comment by JhuoW** [comment]
Thanks for your explanation.
